src/widgets/helper/ResizingTextEdit.cpp

```diff
@@ -30,6 +33,17 @@ ResizingTextEdit::ResizingTextEdit()
     this->installEventFilter(this);
 }
 
+void ResizingTextEdit::editTextChanged()
+{
+    if (getSettings()->messageOverflow.getValue() == 1 &&
+        this->toPlainText().length() > SplitInput::TWITCH_MESSAGE_LIMIT)
+    {
+        QTextCursor cursor = this->textCursor();
+        this->setTextCursor(cursor);
+        cursor.deletePreviousChar();
+    }
+}
```

Comment:
Could this logic just sit in SplitInput's `editTextChanged`?

Reply:
I've moved it over, but could there be cases where this feature is needed in other ResizingTextEdit instances?